osvr_tracked_device.h

```diff
@@ -270,28 +258,60 @@ class OSVRTrackedDevice : public vr::ITrackedDeviceServerDriver
     static void HmdTrackerCallback(void* userdata, const OSVR_TimeValue* timestamp, const OSVR_PoseReport* report);
     const std::string m_DisplayDescription;
     osvr::clientkit::ClientContext& m_Context;
+    osvr::clientkit::DisplayConfig m_DisplayConfig;
     vr::IDriverLog* logger_ = nullptr;
     vr::IServerDriverHost* driver_host_ = nullptr;
     osvr::clientkit::Interface m_TrackerInterface;
-    std::unique_ptr<OSVRDisplayConfiguration> m_DisplayConfiguration;
     vr::DriverPose_t pose_;
     vr::TrackedDeviceClass deviceClass_;
 };
 
-OSVRTrackedDevice::OSVRTrackedDevice(const std::string& display_description, osvr::clientkit::ClientContext& context, vr::IServerDriverHost* driver_host, vr::IDriverLog* driver_log) : m_DisplayDescription(display_description), m_Context(context), driver_host_(driver_host), logger_(driver_log), m_DisplayConfiguration(nullptr), pose_(), deviceClass_(vr::TrackedDeviceClass_HMD)
+OSVRTrackedDevice::OSVRTrackedDevice(const std::string& display_description, osvr::clientkit::ClientContext& context, vr::IServerDriverHost* driver_host, vr::IDriverLog* driver_log) : m_DisplayDescription(display_description), m_Context(context), driver_host_(driver_host), logger_(driver_log), pose_(), deviceClass_(vr::TrackedDeviceClass_HMD)
 {
-    // do nothing
 }
 
 vr::HmdError OSVRTrackedDevice::Activate(uint32_t object_id)
 {
-    // Retrieve display parameters
-    m_DisplayConfiguration = std::make_unique<OSVRDisplayConfiguration>(m_DisplayDescription);
-
+    const std::time_t waitTime = 5; // wait up to 5 seconds for init
     // Register tracker callback
     if (m_TrackerInterface.notEmpty()) {
         m_TrackerInterface.free();
     }
+
+    // ensure context is fully started up
+    logger_->Log("Waiting for the context to fully start up...\n");
+    std::time_t startTime = std::time(nullptr);
+    while (!m_Context.checkStatus()) {
+        m_Context.update();
+        if(std::time(nullptr) > startTime + waitTime) {
+            logger_->Log("Context startup timed out!\n");
```

Comment:
Annoyingly, these log calls only dump to the steamvr client log, not to stdout/stderr, which is annoying for a user who's trying to debug why their HMD isn't working. Should we also print to stderr for fatal errors like these?